fluss-client/src/main/java/com/alibaba/fluss/client/table/writer/AbstractTableWriter.java

```diff
@@ -80,16 +82,22 @@ protected CompletableFuture<Void> send(WriteRecord record) {
         return future;
     }
 
-    protected PhysicalTablePath getPhysicalPath(InternalRow row) {
+    protected PhysicalTablePath getPhysicalPath(
+            InternalRow row, Admin admin, boolean isDynamicCreatePartition) {
         // not partitioned table, return the original physical path
         if (partitionFieldGetter == null) {
             return PhysicalTablePath.of(tablePath);
         } else {
             // partitioned table, extract partition from the row
             String partition = partitionFieldGetter.getPartition(row);
+            ResolvedPartitionSpec resolvedPartitionSpec =
+                    partitionFieldGetter.getResolvedPartitionSpec(row);
```

Comment:
Move this into `if (!isExists && isDynamicCreatePartition) {}` to avoid call every time even if partition exists